### PR TITLE
add required for positional arguments

### LINKED ISSUE #12503: Make positional arguments required in the argument parser
## 🚀 Feature

Make positional arguments required for classes passed into the `add_argparse_args` function.

### Motivation

I use the `add_argparse_args` function not only for the Trainer but also for additional classes required for my training script (e.g. the DataModule). Some arguments of the additional classes are positional and do not have suitable default values. In the current implementation, positional arguments receive a default value of `inspect._empty`, which usually leads to unforeseen errors somewhere further down the line.

### Pitch

In the ptl argument parser utilities, modify the parser such that positional arguments of a class given to the `add_argparse_args` are marked as required in the resulting ArgumentParser.

cc @borda

### PR DESCRIPTION
**Pull request by fschlatt**
add required for positional arguments

## What does this PR do?

Makes positional arguments of a class required in the argument parser returned by `add_argparse_args`.

Fixes #12503

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
great!

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/utilities/argparse.py [hunk 1]
```diff
@@ -266,7 +266,13 @@ def add_argparse_args(
             use_type = _precision_allowed_type
 
         parser.add_argument(
-            f"--{arg}", dest=arg, default=arg_default, type=use_type, help=args_help.get(arg), **arg_kwargs
+            f"--{arg}",
+            dest=arg,
+            default=arg_default,
+            type=use_type,
+            help=args_help.get(arg),
+            required=(arg_default == inspect._empty),
```
Comment (carmocca) [inline pytorch_lightning/utilities/argparse.py:274]: Please also update the `CHANGELOG`, concretely the "Changed" section

Comment (fschlatt) [reply pytorch_lightning/utilities/argparse.py:274]: Done
